sklearn/ensemble/_iforest.py

```diff
@@ -311,7 +312,8 @@ def predict(self, X):
         check_is_fitted(self)
         X = check_array(X, accept_sparse='csr')
         is_inlier = np.ones(X.shape[0], dtype=int)
-        is_inlier[self.decision_function(X) < 0] = -1
+        # is_inlier[self.decision_function(X) < 0] = -1
+        is_inlier[self.decision_function(X) < -1.0e-15] = -1
```

Comment:
Could we use `-np.finfo(float).eps` instead of `-1.0e-15`?

Reply:
Sure, but I had to double it to `-2*np.finfo(float).eps`. Anyway, I have a proper fix for the numerical instability lined up, but in order for it to work, the path length has to be fixed first.